### PR TITLE
Proposal to add context menu to editor tabs

### PR DESCRIPTION
**Pull request by vzhukovskii**
Context menu contains:

1. Close action - closes current selected editor.
2. Close All action - closes all opened editors.
3. Close All but Pinned action - closes all editors that don't have pin.
4. Close Other action - closes all editors but leaves current selected.
5. Reopen Closed Tab action - reopens last closed editor.
6. Pin/Unpin Editor Tab - mark/unmark current selected tab with pin.

Screencast: https://slack-files.com/T02G3VAG4-F0H1D6P47-d3213511d4

@vparfonov @evidolob @skabashnyuk wdyt?

### CODE REVIEW

**Comment by TylerJewell** [comment]
Hi.  Can you capitalize the "But" in menu selection to be consistent with other menus

**Comment by vzhukovskii** [comment]
Done

**Comment by vparfonov** [comment]
ok

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/ContextMenu.java [hunk 1]
```diff
@@ -46,7 +47,7 @@
     private final ActionManager   actionManager;
     private final KeyBindingAgent keyBindingAgent;
 
-    private final PresentationFactory          presentationFactory;
+    protected final PresentationFactory          presentationFactory;
```
Comment (ashumilova) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/ContextMenu.java]: formatter

**Comment by ashumilova** [comment]
+1

**Pull request merged**